server/config-docs.ts

```diff
@@ -350,33 +350,100 @@ const makeDocusaurusCategoryFromEntry = (
   return category;
 };
 
+// Docusaurus doesn't export the types it uses internally for sidebar
+// categories, and these are a little involved, so we'll accept any object.
+// See:
+// https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-plugin-content-docs/src/sidebars/types.ts
+export interface DocusaurusCategory {
+  [propName: string]: unknown;
+}
+
+// categoryDirPattern matches a slug that has been normalized to include a
+// version, or the default version. Examples:
+// - /ver/10.x/installation/
+// - /database-access/introduction/
+const categoryDirPattern = `(/ver/[0-9]+\.x)?/([^/]*)`;
+
+// getIndexPageID infers the Docusaurus page ID of a category index page based
+// on the slugs of pages within the category. The legacy config.json format does
+// not specify the slugs of category index pages within each top-level category
+// so we need to generate this to add links to these pages in the Docusaurus
+// sidebar.
+export const getIndexPageID = (category: NavigationCategory): string => {
+  if (category.entries.length == 0 && !category.generateFrom) {
+    throw new Error(
+      `a navigation category with no generateFrom property must have entries`
+    );
+  }
+
+  // Base the ID on the directory we generated the sidebar from in the legacy
+  // docs site.
+  if (!!category.generateFrom) {
+    return category.generateFrom + "/" + category.generateFrom;
+  }
```

Comment:
I don't think you need the double negation here? you could also do `Boolean(category.generateFrom)` if the optional nature of this field is giving you trouble. i dont mind either way